traits/trait_base.py

```diff
@@ -187,15 +187,15 @@ def enum_default(values):
 
     Parameters
     ----------
-    values : tuple, list or enum.Enum
+    values : tuple, list, collection or enum.Enum
         The collection of valid values for an enum trait.
 
     Returns
     -------
     default : any
         The first valid value, or None if the collection is empty.
     """
-    if isinstance(values, enum.EnumMeta):
+    if isinstance(values, enum.EnumMeta) or is_collection(values):
```

Comment:
At this point it seems as though this line now catches everything we care about, so the other branches here wouldn't ever be exercised.

IOW, we could simply spell the whole function as `return next(iter(values), None)`.

Reply:
Which reminds me: we still need to update the documentation to warn about the particular case of using a `set` as a collection without specifying a default: in that case, the default will be arbitrary.

Reply:
Hmm, seems that defining `__getitem__` is enough to make it a iterable, will change it.

Reply:
Function removed.